man/systemd.exec.xml

```diff
@@ -933,7 +933,9 @@
         (propagation in the opposite direction continues to work).
         This means that this setting may not be used for services
         which shall be able to install mount points in the main mount
-        namespace.</para></listitem>
+        namespace. The /dev namespace will be mounted read-only and 'noexec'.
+        The latter may break old programs which try to set up executable
+        memory by mmap()ing /dev/zero instead of using MAP_ANON.</para></listitem>
```

Comment:
Using Debian codesearch, checking roughly 100 out of 1000 hits for either PROT_EXEC or "/dev/zero", I found that at compilation phase for vast majority of programs, it's checked if MAP_ANON is available and that is preferred. This is a typical example:

https://sources.debian.net/src/libsdl1.2/1.2.15%2Bdfsg1-4/src/stdlib/SDL_malloc.c/?hl=1354#L1354

Some code mention that Linux, BSD and Solaris have provided MAP_ANON and IRIX or other SysV needed the /dev/zero method, so it seems quite old.

The only exceptions where only /dev/zero is used that I found are
v86d
https://sources.debian.net/src/v86d/0.1.10-1/libs/lrmi-0.10/lrmi.c/?hl=143#L143
vbetool
https://sources.debian.net/src/vbetool/1.1-3/x86-common.c/?hl=60#L60
libx86
https://sources.debian.net/src/libx86/1.1%2Bds1-10/x86-common.c/?hl=60#L60

libx86 is a shared library, used by fbterm, read-edid, uswsusp, v86d and vbetool.

At least vbetool and read-edid fails with noexec /dev. They don't seem to work on my computer with exec /dev either. The packages don't ship systemd unit files, so it would be up to system administrator to maintain such files.

Perhaps release notes could mention this change to alert the users of those programs (as systemd units with also PrivateDev=true, not very likely I think).

I'll use the XML tags, thanks.